pyte/screens.py

```diff
@@ -1043,6 +1048,26 @@ def debug(self, *args, **kwargs):
         By default is a noop.
         """
 
+    def scroll_up(self, lines):
+        """Scroll up `lines` lines."""
+        lines_to_scroll = range(self.margins.top, self.margins.bottom + 1)
+        for y in lines_to_scroll:
+            if y + lines in set(lines_to_scroll):
+                self.buffer[y] = self.buffer[y + lines].copy()
+            else:
+                self.buffer[y].clear()
+        self.dirty = set(lines_to_scroll)
```

Comment:
Should update the dirty set rather than replace it.
```suggestion
        self.dirty.update(lines_to_scroll)
```